src/storage/query/QueryBaseProcessor.h

```diff
@@ -34,6 +37,32 @@ struct Bucket {
 
 using OneVertexResp = std::tuple<PartitionID, VertexID, kvstore::ResultCode>;
 
+class ReservoirSampling final {
+public:
+    ReservoirSampling() {
+        rng_.seed(nebula::time::WallClock::fastNowInMicroSec());
+    }
+
+    int64_t sampling() {
+        ++cnt_;
+        if (cnt_ < FLAGS_max_edge_returned_per_vertex) {
+            return cnt_;
+        } else {
+            std::uniform_int_distribution<> dist(0, cnt_);
+            int64_t index = dist(rng_);
```

Comment:
You could use folly::random directly here.

Reply:
I had read it before, its a simple wrapper of PRNG, the code is almost the same as here. So it's not necessary i think.